File: 7kyu/7kyu - Find the capitals.js
// 7kyu - Find the capitals

/* Write a function that takes a single string (word) as argument. 
The function must return an ordered list containing the indexes of all capital letters in the string. */
 
var capitals = function (word) {
    for (var res = [], i = 0; i < word.length; i++)
        if (word[i] == word[i].toUpperCase())
            res.push(i)
    return res
}

q = capitals('CodEWaRs') // [0,3,4,6] 
q

File: 7kyu/7kyu - Find the capitals.py
# 7kyu - Find the capitals

""" Write a function that takes a single string (word) as argument. 
The function must return an ordered list containing the indexes of all capital letters in the string. """


# def capitals(word):
#     return [i for i, c in enumerate(word) if c.upper() == c]

def capitals(word):
    return [i for i, c in enumerate(word) if c.isupper()]


q = capitals('CodEWaRs')  # [0,3,4,6]
q
